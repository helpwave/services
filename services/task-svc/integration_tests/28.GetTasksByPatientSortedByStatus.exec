#!/bin/bash

echo "Exists, has tasks:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "8763d32d-73d1-4e53-b367-b970ee5f6c18"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTasksByPatientSortedByStatus

echo "Exists, has no tasks:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "5f00d59a-72b1-4eb6-9817-5a3aec7476ef"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTasksByPatientSortedByStatus

echo "Does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"patient_id": "6f00d59a-72b1-4eb6-9817-5a3aec7476ef"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTasksByPatientSortedByStatus
