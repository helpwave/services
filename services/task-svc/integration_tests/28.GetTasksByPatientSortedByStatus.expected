Exists, has tasks:
{
  "todo": [
    {
      "id": "0a4b4616-ba16-432a-ba50-55b1b58c8a88",
      "name": "Revision of Autol Sub in L Thumb Phalanx, Extern Approach",
      "description": "Introduction of Oxazolidinones into Lymphatics, Percutaneous Approach",
      "assignedUserId": "df363387-7774-4592-b01c-2b765abaa86a",
      "patientId": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
      "public": true,
      "dueAt": "2024-04-07T18:21:24Z",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421"
    }
  ],
  "inProgress": [
    {
      "id": "ceffe1b1-84f6-42c7-b058-dc2897509f49",
      "name": "Introduce of Radioact Subst into Fem Reprod, Perc Approach",
      "assignedUserId": "3a82ae88-c43a-4ada-8659-9c5f372cb48d",
      "patientId": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
      "dueAt": "2023-07-02T16:48:52Z",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421",
      "subtasks": [
        {
          "id": "2ca50d51-0d75-4ee4-9d91-f9fe4561402b",
          "name": "Revision or closure of vesicostomy",
          "createdBy": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758"
        }
      ]
    }
  ],
  "done": [
    {
      "id": "b025f9bb-7852-45c2-82a6-bfee3adcbddd",
      "name": "Removal of Drain Dev from Vagina \u0026 Cul-de-sac, Perc Approach",
      "description": "Supplement Left Nipple with Autologous Tissue Substitute, Via Natural or Artificial Opening Endoscopic",
      "assignedUserId": "00000000-0000-0000-0000-000000000000",
      "patientId": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
      "public": true,
      "dueAt": "2024-06-12T00:04:24Z",
      "createdBy": "674217fc-05a6-4eec-ad7e-8a2d0dc8caa9",
      "subtasks": [
        {
          "id": "5116c01d-5f0d-46e9-bcd7-0d2127661dc7",
          "name": "Insertion of intercostal catheter for drainage",
          "done": true,
          "createdBy": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758"
        },
        {
          "id": "cc60cc5f-028c-46be-80b9-6c4f7ea4ec65",
          "name": "Mediastinal pneumogram",
          "createdBy": "3e795d31-1912-4b09-8e5a-fd62a2461a8c"
        }
      ]
    },
    {
      "id": "a2f04cb2-4587-48ad-89ee-a01256a80d7c",
      "name": "Reposition Left Hand Tendon, Open Approach",
      "description": "Dilation of Left Radial Artery with Four or More Drug-eluting Intraluminal Devices, Percutaneous Approach",
      "assignedUserId": "5db6cc86-bbf1-4fa6-bbe8-a8cdea869758",
      "patientId": "8763d32d-73d1-4e53-b367-b970ee5f6c18",
      "dueAt": "2023-11-04T07:55:09Z",
      "createdBy": "df363387-7774-4592-b01c-2b765abaa86a"
    }
  ]
}
Exists, has no tasks:
{}
Does not exist:
{}
